### PR TITLE
MinGW build fixes

### PR DESCRIPTION
**Pull request by madebr**
These are build fixes to get isle building (further) with a modern mingw toolchain.

The only errors left are those due to taking the address of rvalues.
[These can be fixed easily](https://github.com/madebr/isle/commit/9dde9e7f9d3a2775b565a2fa9504e04afc78bf89), but it decreases the matching percentage.
Therefore, not part of this pr.

### CODE REVIEW

File: LEGO1/mxobjectfactory.cpp [hunk 1]
```diff
@@ -28,16 +28,14 @@ MxObjectFactory::MxObjectFactory()
 // OFFSET: LEGO1 0x100b12c0
 MxCore *MxObjectFactory::Create(const char *p_name)
 {
-  MxCore* object = NULL;
```
Comment (foxtacles) [inline LEGO1/mxobjectfactory.cpp:31]: It looks like there is a a major accuracy regression with this one

File: LEGO1/mxhashtable.h [hunk 2]
```diff
@@ -9,6 +9,11 @@
 #define HASH_TABLE_OPT_EXPAND_ADD         1
 #define HASH_TABLE_OPT_EXPAND_MULTIPLY    2
 
+template <class T>
+class MxHashTableCursor;
+template <class T>
+class MxList;
```
Comment (foxtacles) [inline LEGO1/mxhashtable.h:15]: Is it supposed to be `MxHashTable`?

Comment (madebr) [reply LEGO1/mxhashtable.h:15]: Only `MxHashTableCursor` is needed here. I got confused by the missing forwarded declarations in `mxlist.h`.

File: LEGO1/mxobjectfactory.cpp (hunk 1 above)
Comment (madebr) [reply LEGO1/mxobjectfactory.cpp]: I reverted the change. It was not needed for mingw compatibility.

**Review by foxtacles — APPROVED** [review]
Looks good!

**Pull request merged**